data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/SubListExpressionFunction.java

```diff
@@ -0,0 +1,53 @@
+package org.opensearch.dataprepper.expression;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import org.opensearch.dataprepper.model.event.Event;
+
+import javax.inject.Named;
+
+@Named
+public class SubListExpressionFunction implements ExpressionFunction {
+
+	@Override
+	public String getFunctionName() {
+		return "subList";
+	}
+
+	@Override
+	public Object evaluate(List<Object> args, Event event, Function<Object, Object> convertLiteralType) {
+		if (args.size() != 3) {
+			throw new IllegalArgumentException("subList() takes 3 arguments");
+		}
+		if (!(args.get(0) instanceof String)) {
+			throw new IllegalArgumentException("subList() takes 1st argument as string type");
+		}
+		if (!(args.get(1) instanceof Integer) || !(args.get(2) instanceof Integer)) {
```

Comment:
Did you run data prepper end to end to test this function? I pulled your PR and it looks like Integers can't be passed as arguments to the functions due to the grammar. I think we will need these to be strings for now and we can throw illegal argument exception if we can't parse the string to int.

```
  - add_entries:
        entries:
          - key: "result"
            value_expression: 'subList(/my_list, "0", "1")'
```